lib/rotp/totp.rb

```diff
@@ -47,6 +47,30 @@ def verify_with_drift(otp, drift, time = Time.now)
       times.any? { |ti| verify(otp, ti) }
     end
 
+    # Verifies the OTP passed in against the current time OTP
+    # and adjacent intervals up to +drift+.  Excludes OTPs
+    # from prior_time and earlier.  Returns time value of
+    # matching OTP code for use in subsequent call.
+    # @param [String] otp the OTP to check against
+    # @param [Integer] drift the number of seconds that the client
+    #     and server are allowed to drift apart
+    # @param [Integer] time value of previous match
+    def verify_with_drift_and_prior(otp, drift, prior_time = nil, time = Time.now)
+      # calculate normalized bin start times based on drift
```

Comment:
I think there's value in having two separate methods.

I like being able to call `verify_with_drift_and_prior` with only three arguments, allowing the fourth (`time`) to default.   But modifying the existing method to support that would require changing the position of the `time` argument and that would likely break existing code.  Adding `prior_time` as an optional fourth argument would not break existing code, but would make defaulting the `time` argument considerably less intuitive.

What I could do is to change the definition of `verify_with_drift` to simply pass its arguments to `verify_with_drift_and_prior` since they should behave identically.  I originally kept the implementations separate to try and make this PR less controversial (by avoiding changing existing code).  But if @mdp prefers, I'm happy to take the bolder approach.